apps/api/src/controllers/applicationController.ts

```diff
@@ -352,3 +352,67 @@ export const requestApplicationRevisionsByRep = async ({
 		return failure('An error occurred while processing the request.', error);
 	}
 };
+
+export const closeApplication = async ({
+	applicationId,
+	requesterId,
+	isDacMember = false,
+}: {
+	applicationId: number;
+	requesterId: string;
+	isDacMember?: boolean;
+}) => {
+	try {
+		const database = getDbInstance();
+		const service: ApplicationService = applicationSvc(database);
+		const result = await service.getApplicationById({ id: applicationId });
+
+		if (!result.success) {
+			return result;
+		}
+
+		const application = result.data;
+		const appStateManager = new ApplicationStateManager(application);
+
+		// Check if application is already closed
+		if (appStateManager.state === ApplicationStates.CLOSED) {
+			return failure('Application is already closed.', 'StateConflict');
+		}
+
+		// Authorization logic based on state
+		const isApplicant = application.user_id === requesterId;
+
+		const canCloseApplication = isDacMember && appStateManager.state === ApplicationStates.DAC_REVIEW;
+
+		if (!isApplicant && !canCloseApplication) {
```

Comment:
This should be `||`, `if not applicant OR cannot close application`

Reply:
also `canCloseApplication` might be good to rename as `canCloseDacApplication` because it is actually pretty specific

Reply:
OK I reviewed the commit/discussion about this. What's missing here is the application state.

```
	if (!isApplicant && applicationState ===  DRAFT || applicationState ===  INSTITUTIONAL_REP_REVIEW ) {
	 return failure 
	 } else if ((!isApplicant || !canCloseDacApplication)) { 
  	return failure
	}
}
```

Reply:
edit: `if (!isApplicant || !canCloseApplication) { return failure ... }` covers all state use cases, so update `&&` to `||` here